app/controllers/api/base_controller.rb

```diff
@@ -278,6 +278,11 @@ def find_nested_object
 
     def not_found_if_nested_id_exists
       allowed_nested_id.each do |obj_id|
+        # this method does not reliably work when you have multiple parameters and some of them can be nil
+        # find_nested_object in such case returns nil (since org and loc can be nil for any context),
+        # but it detects other paramter which can have value set
+        # therefore we always skip these
+        next if [ 'organization_id', 'location_id' ].include?(obj_id)
```

Comment:
I think the method `find_optional_nested_object` should know, which param was used for searching for the resource, and then checking just on presence of that resource. I'm ok to accept this change only, if there will going to be additional PR that is going to fix this better way. This is too big of a hack for me.

Reply:
This never worked before reliably. If you look how much we use @nested_obj in our code, it shouldn't be a big concern. We can't tell what parameter was supposed to be used if the nested obj is nil, so the whole logic would need to be reimplemented. Also this code supports only one parameter for nesting while controllers specify multiple allowed nested ids. We should later remove org and loc from allowed nested ids from all controllers, but that is potentially change for all plugins. We should redesign how we use nested routes in API v3. And ideally drop this code completely.